### PR TITLE
SCMP review

### PR DESCRIPTION
**Pull request by nicorusti**
Follow up to #49.  Contributes to #36 

[Diff with last submission](https://author-tools.ietf.org/api/iddiff?doc_1=draft-dekater-scion-dataplane&url_2=https://scionassociation.github.io/scion-dp_I-D/scmp_review/draft-dekater-scion-dataplane.txt)

### CODE REVIEW

**Comment by nicorusti** [comment]
I had an overall look at SCMP, I have some questions: 
- Shall we also describe echo messages at this point?

**Review by nicorusti — COMMENTED** [review]
@jiceatscion what do you think about my comments?  I feel SCMP still needs some work..

File: draft-dekater-scion-dataplane.md [hunk 1]
```diff
@@ -1393,14 +1389,11 @@ This specification defines the message formats for the following SCMP messages:
...
 
 Type values 100, 101, 200, and 201 are reserved for private experimentation.
-They are not intended for general use. Any wide-scale and/or uncontrolled usage
-should obtain a real allocation.
 
-Type values 127 and 255 are reserved for future expansion of in case of a
-shortage of type values.
+All other values are reserved for future use.
 
 ## Checksum Calculation
```
[the remark this answers is not in the document]
Comment (oncilla) [reply draft-dekater-scion-dataplane.md:1398]: This is taken word for word from https://www.rfc-editor.org/rfc/rfc4443.html#section-2.3

File: draft-dekater-scion-dataplane.md [hunk 2]
```diff
@@ -1346,17 +1344,17 @@ The messages have the following general format:
...
+- `Code`: provides additional granularity to the SCMP type.
 
-*Checksum* is used to detect data corruption.
+- `Checksum`: is used to detect data corruption.
 
-*InfoBlock* is an optional field of variable length. The format is dependent on the message type.
+- `InfoBlock` is an optional field of variable length. The format is dependent on the message type.
 
-*DataBlock* is an optional field of variable length. The format is dependent on the message type.
+- `DataBlock` is an optional field of variable length. The format is dependent on the message type.
```
[the remark this answers is not in the document]
Comment (jiceatscion) [reply draft-dekater-scion-dataplane.md]: I thought I'd removed it from the text as well. The two blocks description serve no purpose. I've deleted the reference.

File: draft-dekater-scion-dataplane.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiceatscion) [reply draft-dekater-scion-dataplane.md:1398]: What I plagiarized was actually this: https://docs.scion.org/en/latest/protocols/scmp.html#checksum-calculation :-)

[the remark this answers is not in the document]
Comment (oncilla) [reply draft-dekater-scion-dataplane.md:1396]: Yes, and now guess where that was plagiarized from :)

[the remark this answers is not in the document]
Comment (nicorusti) [reply draft-dekater-scion-dataplane.md:1396]: Hahaha.. well, if this wording is also used in in [rfc4443.html#section-2.3](https://www.rfc-editor.org/rfc/rfc4443.html#section-2.3), let's leave it!

**Comment by jiceatscion** [comment]
>     * Shall we also describe echo messages at this point?
I see no reason to.

>     * Section "## Processing Rules" has wording that could be imprived
Anything specific?

Working on the rest.

**Comment by jiceatscion** [comment]
>       * Why 1232 bytes ? What if there is a lower MTU for whatever reason? --> Check IPv6 minimum MTU 

Here's sortta why:
IPV6 requires a minimum 1280 bytes MTU. The IPV6 HDR is 40 bytes (if no extensions). Since we send SCMP packets over UDP, that's another 8. 1280-40-8 = 1232. Now, that only makes sense over V6 links. I presume that for V4 links the author assumed we would rely on either UDP fragmentation or non-stupid MTU. It is widely accepted that no IPV4 router in use today has links with MTU below 1280. So, it makes sense to use 1280 from V6 as the only floor that maters.

**Comment by jiceatscion** [comment]
I added an explanation regarding 1232. Also added ping and fixed a cpl of table/figure issues.

**Comment by nicorusti** [comment]
Hi @jiceatscion , thanks a lot for your previous changes! 
I reworded the section about BDF and link failures, and I tried to make it a bit more concise. I also edited a couple of SCMP header figures. Could you have a quick look, and if ok, feel free to merge this :)

**Comment by jiceatscion** [comment]
I disagree with merging link failures in SCMP. I think it was like that in some earlier iteration and it ends-up messy. SCMP is for many things besides link failure. Link failure is about more than SCMP (there's BFD too). So, neither is a proper subset of the other.

**Comment by jiceatscion** [comment]
I made a coupld of very minor changes, removed a stray preposition here, added a missing article there... still GH wants another stamp of approval.

**Review by jiceatscion — APPROVED** [review]
I made a couple of very minor changes, removed a stray preposition here, added a missing article there...

**Pull request merged**